src/main/java/org/assertj/core/api/exception/package-info.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+@NonNullApi
+@NonNullFields
+package org.assertj.core.api.exception;
```

Comment:
To explain how this works:
I create some meta-annotations (@Nullable, @NonNullApi, etc) The meta annotations are annotated with the JSR 305 annotation, so user won't have to worry about them and the compiler can infer their use.

Then by annotating each `package-info.java` we set the API default to be `NonNull`, that way we only have to add `@Nullable` as it is appropriate.